lib/descriptors/MessageDescriptor.js

```diff
@@ -136,6 +140,37 @@ MessageDescriptor.prototype.getFieldNames = function () {
 }
 
 
+MessageDescriptor.prototype.getNextOneofIndex = function () {
+  return this._oneofs.length
+}
+
+
+MessageDescriptor.prototype.addOneof = function (oneof) {
```

Comment:
I was considering that as well, but it seemed to go against the patterns of the other descriptors. It would be simpler though... let's see.